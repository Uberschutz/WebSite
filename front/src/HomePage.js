import React, { Component } from 'react';
import './styles/bootstrap.css';

import Header from "./Header";
import './styles/HomePage.css';
import canvas from './assets/canvas.png';

class HomePage extends Component {
    render() {
        return (
            <div>
                <Header/>
                <img src={canvas} alt="canvas" className="responsive-image"/>
            </div>
        )
    }
}

export default HomePage;