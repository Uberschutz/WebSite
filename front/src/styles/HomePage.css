.uber-color {
    background-color: cornflowerblue;
    color: white!important;
}

.responsive-image {
    position: relative;
    max-width: 100%;
}