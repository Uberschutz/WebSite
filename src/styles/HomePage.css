.Home {
    text-align: center;
    align-items: center;
    align-content: center;
    justify-items: center;
    justify-content: center;
}

.uber-color {
    background-color: cornflowerblue;
    color: white!important;
}

.uber-color2 {
    background-color: black;
    color: white!important;
}

.responsive-image {
    position: relative;
    max-width: 100%;
}

.text-span {
    position: absolute;
    font-size: 17pt;
    top: 300px;
    left: 130px;
    color: white;
}

.button-footer {
    margin: 15px;
}

.background_1 {
    background-color: #1d2124;
    color: white;
}

.italic-txt {
    font-style: italic;
}

.h7-font {
    font-weight: bold;
}

.h6-font {
    font-size: 120%;
}

.txt-intro {
    margin-top: 7%;
    margin-left: 7%;
    text-align: left!important;
}

.rappel-info {
    font-size: 150%;
}

.clickable-flag {
    cursor: pointer;
}

.touched {
    opacity: 0.8;
}