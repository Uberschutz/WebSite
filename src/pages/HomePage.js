import React, { Component } from 'react';
import '../styles/bootstrap.css';
import '../styles/HomePage.css';

import canvas from '../assets/canvas2.png';
import whois from '../assets/CacherPseudo.jpg';
import software from '../assets/logiciel.png';
import navigation from '../assets/navigation.png';
import blocking from '../assets/bloquant.png';

const content = require('../assets/text');

class HomePage extends Component {
	constructor(props) {
		super(props);

		this.state = {
			lang: 'fr'
		};
	}

	componentDidMount() {
		this.setState({
			lang: this.props.lang
		})
	}

	componentDidUpdate(prevProps, prevState, snapshot) {
		console.log(this.props.lang);
	}

	displayContent(content) {
		// return(<div dangerouslySetInnerHTML={ {__html: content} }/>)
		return(content)
	}

	render() {
		let i = 0;
        return (
            <div>
                <img src={canvas} alt="canvas" className="responsive-image"/>
                <span className="text-span">
	                {this.displayContent(content.filter(obj => obj.lang === this.props.lang)[0].pages.home[i++])}
                    <br/> {this.displayContent(content.filter(obj => obj.lang === this.props.lang)[0].pages.home[i++])}
                    <br/> {this.displayContent(content.filter(obj => obj.lang === this.props.lang)[0].pages.home[i++])}
                </span>
                <div className="row responsive-image Home">
                    <div>
                        <br/><br/>
                        <img src={whois} alt="whois"/>
                    </div>
                    <div className="txt-intro">
                        <h3 className="h7-font">
	                        {this.displayContent(content.filter(obj => obj.lang === this.props.lang)[0].pages.home[i++])}
                        </h3>
                        <h6 className="h6-font">
	                        {this.displayContent(content.filter(obj => obj.lang === this.props.lang)[0].pages.home[i++])}
                            <br/>
	                        {this.displayContent(content.filter(obj => obj.lang === this.props.lang)[0].pages.home[i++])}
                        </h6>
                        <h8 className="italic-txt">
	                        {this.displayContent(content.filter(obj => obj.lang === this.props.lang)[0].pages.home[i++])}
                        </h8>
                    </div>
                </div>
                <div className="Home responsive-image">
                    <br/><br/>
                    <h5 className="rappel-info h7-font">
	                    {this.displayContent(content.filter(obj => obj.lang === this.props.lang)[0].pages.home[i++])}
                    </h5>
                    <br/><br/>
                    <div className="row responsive-image">
                        <div className="col">
                            <img src={software} alt="software"/>
                            <br/><br/>
                            <span className="rappel-info">
			                    {this.displayContent(content.filter(obj => obj.lang === this.props.lang)[0].pages.home[i++])}
                            </span>
                        </div>
                        <div className="col responsive-image">
                            <img src={navigation} alt="navigation"/>
                            <br/><br/>
                            <span className="rappel-info">
        	                    {this.displayContent(content.filter(obj => obj.lang === this.props.lang)[0].pages.home[i++])}
		                    </span>
                        </div>
                        <div className="col responsive-image">
                            <img src={blocking} alt="blocking"/>
                            <br/> <br/>
                            <span className="rappel-info">
        	                    {this.displayContent(content.filter(obj => obj.lang === this.props.lang)[0].pages.home[i++])}
		                    </span>
                        </div>
                    </div>
                    <br/>
                    <h7 className="rappel-info">
	                    {this.displayContent(content.filter(obj => obj.lang === this.props.lang)[0].pages.home[i])}
                    </h7>
                    <br/><br/><br/>
                </div>
            </div>
        )
    }
}

export default HomePage;