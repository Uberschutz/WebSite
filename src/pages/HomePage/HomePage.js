import React, { Component } from 'react';
import '../../styles/bootstrap.css';
import '../../styles/HomePage.css';

import ReactGA from 'react-ga';

import canvas from '../../assets/canvas2.png';
import whois from '../../assets/CacherPseudo.jpg';
import software from '../../assets/icons8-boite-à-logiciel-64.png'
import navigation from '../../assets/icons8-carte-au-trésor-150.png'
import blocking from '../../assets/icons8-cadenas-100.png'

import { displayContent } from '../../utils/translationDisplay';

class HomePage extends Component {
	constructor(props) {
		super(props);

		this.state = {
			lang: 'fr'
		};
    };

	componentDidMount() {
		if (process.env.ANALYTICS === 'true') {
			ReactGA.pageview(window.location.pathname + window.location.search);
		}
		if (this.props.base) {
			const {base: {language}} = this.props;
			// console.log(this.props, 'in home');
			// const { base: { token }} = this.props;
			if (this.state.lang !== language) {
				this.setState({
					lang: language
				})
			}
		}
	}

	componentDidUpdate(prevProps, prevState, snapshot) {
		// console.log(prevProps, this.props);
		// this.props.base.language !== prevProps.base.language && this.setState({lang: this.props.base.language}, () => console.log('re'));
		if (this.props.base && this.props.base.language !== this.state.lang) {
			// this.props.base.language !== prevState.language && this.setState({lang: this.props.base.language}, () => console.log('re'));
			this.setState({lang: this.props.base.language})
		}
	}

	// shouldComponentUpdate(nextProps, nextState, nextContext) {
	// 	console.log('next', nextProps)
	// 	return (this.props === nextProps);
	// }

	render() {
		let i = 0;
        return (
            <div>
                <img src={canvas} alt="canvas" className="responsive-image"/>
                <span className="text-span">
	                {displayContent(this.state.lang, i++, 'home')}
                    <br/> {displayContent(this.state.lang, i++, 'home')}
                    <br/> {displayContent(this.state.lang, i++, 'home')}
                </span>
                <div className="row responsive-image Home">
                    <div>
                        <br/><br/>
                        <img src={whois} alt="whois" className="responsive-image"/>
                    </div>
                    <div className="txt-intro responsive-image">
                        <h3 className="h7-font">
	                        {displayContent(this.state.lang, i++, 'home')}
                        </h3>
                        <h6 className="h6-font">
	                        {displayContent(this.state.lang, i++, 'home')}
                            <br/>
	                        {displayContent(this.state.lang, i++, 'home')}
                        </h6>
                        <h8 className="italic-txt">
	                        {displayContent(this.state.lang, i++, 'home')}
                        </h8>
                    </div>
                </div>
                <div className="Home responsive-image">
                    <br/><br/>
                    <h5 className="rappel-info h7-font">
	                    {displayContent(this.state.lang, i++, 'home')}
                    </h5>
                    <br/><br/>
                    <div className="row responsive-image">
                        <div className="col">
                            <img src={software} alt="software"/>
                            <br/><br/>
                            <span className="rappel-info">
			                    {displayContent(this.state.lang, i++, 'home')}
                            </span>
                        </div>
                        <div className="col responsive-image">
                            <img src={navigation} alt="navigation"/>
                            <br/><br/>
                            <span className="rappel-info">
        	                    {displayContent(this.state.lang, i++, 'home')}
		                    </span>
                        </div>
                        <div className="col responsive-image">
                            <img src={blocking} alt="blocking"/>
                            <br/> <br/>
                            <span className="rappel-info">
        	                    {displayContent(this.state.lang, i++, 'home')}
		                    </span>
                        </div>
                    </div>
                    <br/>

                    <div className="">
                        <h7 className="rappel-info responsive-image">
                            {displayContent(this.state.lang, i, 'home')}
                            <br/>
                        </h7>
                        <br/>
                    </div>


                </div>
            </div>
        )
    }
}

export default HomePage;
