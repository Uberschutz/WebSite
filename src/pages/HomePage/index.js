import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import * as actions from '../../actions'
import HomePage from './HomePage'

const mapStateToProps = state => ({
	base: state.base,
})

const mapDispatchToProps = dispatch => ({
	getServiceData: bindActionCreators(actions.getServiceData, dispatch),
	sendServiceData: bindActionCreators(actions.sendServiceData, dispatch),
	saveAuthToken: bindActionCreators(actions.saveAuthToken, dispatch),
	subscribeToService: bindActionCreators(actions.subscribeToService, dispatch),
	updateFill: bindActionCreators(actions.updateFill, dispatch),
	updateActionFill: bindActionCreators(actions.updateActionFill, dispatch),
	toggleAREA: bindActionCreators(actions.toggleAREA, dispatch),
})

const HomePageContainer = connect(mapStateToProps, mapDispatchToProps)(HomePage)

export default HomePageContainer