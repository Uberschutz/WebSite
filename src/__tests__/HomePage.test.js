import React from 'react';
import ReactDOM from 'react-dom';
import HomePage from '../pages/HomePage/HomePage';

import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow, render, mount } from 'enzyme';
// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

it('renders Home without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<HomePage/>, div);
    expect(div).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
});

it('renders Home in english', function () {
	const div = document.createElement('div');
	const homeComponent = shallow(<HomePage/>);
	homeComponent.setProps({base: { language: 'fr'}});
	// homeComponent.update();
	homeComponent.setState({ lang: 'en' });
	ReactDOM.render(homeComponent, div);
	expect(div).toMatchSnapshot();
	ReactDOM.unmountComponentAtNode(div);
});
