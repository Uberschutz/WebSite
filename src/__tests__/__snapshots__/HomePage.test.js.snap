// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home without crashing 1`] = `
<div>
  <div>
    <img
      alt="canvas"
      class="responsive-image"
      src="canvas2.png"
    />
    <span
      class="text-span"
    >
      Internet :
      <br />
       
      êtes-vous conscients des dangers
      <br />
       
      que rencontrent vos enfants ?
    </span>
    <div
      class="row responsive-image Home"
    >
      <div>
        <br />
        <br />
        <img
          alt="whois"
          src="CacherPseudo.jpg"
        />
      </div>
      <div
        class="txt-intro"
      >
        <h3
          class="h7-font"
        >
          Notre solution :
        </h3>
        <h6
          class="h6-font"
        >
          Pour une navigation saine, utilisez Überschutz.
          <br />
          Ce logiciel non bloquant vous préviendra lorsque votre enfant se met en danger sur Internet.
        </h6>
        <h8
          class="italic-txt"
        >
          Suicide, drogue, anorexie, harcèlement, contenus pornographiques ... Vous n'imaginez pas à quoi il est confronté.
        </h8>
      </div>
    </div>
    <div
      class="Home responsive-image"
    >
      <br />
      <br />
      <h5
        class="rappel-info h7-font"
      >
        En quelques mots :
      </h5>
      <br />
      <br />
      <div
        class="row responsive-image"
      >
        <div
          class="col"
        >
          <img
            alt="logiciel"
            src="logiciel.png"
          />
          <br />
          <br />
          <span
            class="rappel-info"
          >
            Un logiciel
          </span>
        </div>
        <div
          class="col responsive-image"
        >
          <img
            alt="navigation"
            src="navigation.png"
          />
          <br />
          <br />
          <span
            class="rappel-info"
          >
            Non bloquant
          </span>
        </div>
        <div
          class="col responsive-image"
        >
          <img
            alt="bloquant"
            src="bloquant.png"
          />
          <br />
           
          <br />
          <span
            class="rappel-info"
          >
            Non intrusif
          </span>
        </div>
      </div>
      <br />
      <h7
        class="rappel-info"
      >
        Pour protéger votre enfant et discuter avec lui lorsqu'une situation compliquée se présente.
      </h7>
      <br />
      <br />
      <br />
    </div>
  </div>
</div>
`;
